src/org/zaproxy/zap/authentication/AuthenticationHelper.java

```diff
@@ -40,7 +51,21 @@ public static void notifyOutputAuthSuccessful() {
 		}
 	}
 
-	public HttpState getCorrespondingHttpState() {
+	public static void notifyOutputAuthFailure(HttpMessage msg) {
+		// Always record stats
+		try {
+			Stats.incCounter(SessionStructure.getHostName(msg), AUTH_FAILURE_STATS);
+		} catch (URIException e) {
+			// Ignore
+		}
+		// Let the user know it worked
+		if (View.isInitialised()) {
+			View.getSingleton().getOutputPanel()
+					.append(Constant.messages.getString("authentication.output.failure") + "\n");
+		}
+	}
+
+	public HttpState getCorrespondingHttpState(HttpMessage msg) {
```

Comment:
Was this change on purpose? The parameter is not being used.
Same comment as above regarding binary compatibility.